runtime/curl.cpp

```diff
@@ -648,6 +1014,66 @@ bool f$curl_setopt(curl_easy easy_id, int64_t option, const mixed &value) noexce
   return false;
 }
 
+constexpr int64_t CURLSETOPT_HEADERFUNCTION = 210000;
+constexpr int64_t CURLSETOPT_WRITEFUNCTION = 210001;
+
+bool curl_setopt_fn_header_write(curl_easy easy_id, int64_t option, curl::on_write_callable callable) noexcept {
+  if (auto *easy_context = get_context<EasyContext>(easy_id)) {
+    switch (option) {
+      case CURLSETOPT_HEADERFUNCTION:
+        easy_context->write_header_handler.callable = callable;
```

Comment:
## The process of installing the user's callable objects when calling `curl_setopt()`
After successful redirection by the compiler of the `curl_setopt()` function to the function from `kphp_internal.txt` the callable object is saved to the appropriate handler. In this case, the mode of operation of the corresponding curl function (`method` field) is set to the value `KPHP_CURL_USER`.